app/jobs/send_payments_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SendPaymentsJob < ApplicationJob
+  queue_as :default
+
+  def perform(payment_ids)
+    payments = Payment.where(id: payment_ids)
+
+    ASP::Request
+      .with_payments(payments, ASP::Entities::Fichier)
+      .send!(ASP::Server)
```

Comment:
Cette interface est très compliquée à comprendre, j'ai beau faire des aller retours je ne suis pas sûr de comprendre.

Est-ce qu'il y a un ticket qui traîne avec l'explication de l'algo ? J'ai l'impression qu'en réalité c'est pas si compliqué et qu'un bon vieux chain of command ferait le taff de manière claire non ?

Reply:
En gros si je comprends bien:

1. On créer un model request asp en y attachant les paiements 
2. On envoie les fichiers sur le serveur asp
3. On marque les paiements comme process

Si c'est le cas, ça me semble assez straightforward de faire quelque chose de plus simple en séquenciel, avec des noms plus explicites.

Reply:
c'est une requêtes à l'ASP via un `ASP::Server` pour un groupe de paiements. Ils sont formattés avec la classe `ASP::Entities::Fichier` qui permet de générer le XML. Je suis pas très satisfait de la tête que ça a avec la dependency injection mais c'était pour pouvoir tester vite et bien.

Reply:
```ruby
def perform
  create_asp_request_linked_to_payments
  send_paiements_to_asp_server
  mark_payments_as_processed!
end
```

J'aime bien utilisé https://github.com/collectiveidea/interactor pour ce genre de chose histoire de tout tester en isolation.

De manière général si je passe plus de 2s à comprendre une ligne je la challenge. Là ça a l'avantage d'être crystal clear.

Reply:
genre 

```ruby
file = make_xml(payments)

request.create(file)

server.send_file(file)

payments.each(&:process!)
```

?

Reply:
Oui par exemple. Après c'est une façon de faire, là juste je trouve que ton interface est assez compliqué à comprendre et très technique (genre passer des classes dans les méthodes c'est un peu brainfuck, est-ce qu'il y a besoin de cette flexibilité/complexité ?))

Reply:
c'est juste de la dependency injection, je suis pas habitué non plus mais effectivement ça rend l'interface plus testable. Je vais voir si il y a mieux à faire, je pense que le `send